File: API Server/handlers/delete-word.js
module.exports = function (req, res, config, redis) {
  var username = req.query.username, word = req.query.word;

  function p (str) { return [config.prefix, username, str].join('.'); };
  function s (str) { return [str, word].join('.'); };
  redis.deleteWord([p('wordset'), p('quiz'), p(s('word')), p(s('star'))], [word], function (err, reply) {
    if (err)
      return res.returnJSON(500, config.errors[500]);
    if (reply == null)
      return res.returnJSON(200, config.errors.word_not_existed);
    res.returnJSON(200, 'succeed');
  });
};

File: API Server/lua/deleteWord.lua
-- deleteWord.lua (wordset, quiz, word, star) (word)

if redis.call('EXISTS', KEYS[1]) == 0 then
  return false
end

redis.call('SREM', KEYS[1], ARGV[1])
redis.call('DEL', KEYS[4])
redis.call('ZREM', KEYS[2], ARGV[1])
redis.call('DEL', KEYS[3])

return true
